src/network.py

```diff
@@ -21,11 +21,19 @@ class Unit:
 
     def __init__(self, w, b, act):
         """
-        Constructor
+        Unit's constructor
         :param w: vector of weights
         :param b: bias (number)
         :param act: activation function --> 'Function' obj (see 'functions.py')
         """
+        if hasattr(w, '__iter__'):
```

Comment:
Questi controlli si possono eliminare perché l'unico modo con cui inizializziamo i pesi e il bias delle units è tramite lo script [weights_inits.py](https://github.com/AlexPasqua/NNs-from-scratch/blob/constructors_check/src/weights_inits.py), che ritorna dati corretti (cioè numeri, non ritornerà stringhe ecc).

L'unico problema possibile che mi viene in mente è sulle dimensioni degli array. Cioè noi diciamo a `weights_inits` di ritornare `n` pesi, mentre in realtà ne servono `m`. In tal caso, siccome noi facciamo un assegnamento del tipo `self.__w = w`, in caso di mismatching shapes verrà lanciato un errore in automatico, quindi non serve che lo lanciamo anche noi.

Che ne pensi?


In caso va eliminata [questa sezione](https://github.com/AlexPasqua/NNs-from-scratch/blob/constructors_check/src/network.py#L29-L36)